src/Type.cpp

```diff
@@ -519,6 +538,30 @@ Type::get_all_ok_struct_union_types(vector<Type *> &ok_types, bool no_const, boo
 	}
 }
 
+bool 
+Type::if_struct_will_have_assign_ops()
+{
+	bool hasAssignOps = false;
+	// randomly choose if the struct will have assign operators (for C++):
+	if (CGOptions::lang_cpp()){
+		hasAssignOps = rnd_flipcoin(RegularVolatileProb);
```

Comment:
More or less. We need those assign ops if we want to use the struct as a volatile, and we need to generate them with some probability. So re-using volatile prob seemed right to me.